Static/DataSource.swift

```diff
@@ -127,9 +127,9 @@ public class DataSource: NSObject {
         return nil
     }
 
-    private func refreshTableSections(oldSections: [Section]? = nil) {
+    private func refreshTableSections(oldSections: [Section] = []) {
         guard let tableView = tableView else { return }
-        guard let oldSections = oldSections else {
+        guard !oldSections.isEmpty else {
```

Comment:
Thanks for the PR @colinhumber ,

Given the move from Optional to isEmpty check, why the hard reload?

If moving from old to new, would it be beneficial to fall through to the code below no matter what (Given the newly assumed `[]`)

cc: @hyperspacemark you might have some context on this?

Reply:
That's a good point. Having the default value as `nil` instead of `[]` could help differentiate between the first time the table view is loaded (`oldSections == nil`) and we'd want a hard reload vs subsequent changes where we may have removed all sections and are inserting new ones and would want those animated.

Thoughts?

Reply:
Given sections is non-nil (thus nil doesn't appear an option), I'm thinking all of these can just fall through to the add/remove/update animation rather than defaulting to reloadData(). Unless I'm missing some historical context here.

Reply:
The more I think about it, I think having the distinction between the table view being loaded for the first time (oldSections = nil) and the sections changing (oldSections = non-nil) makes sense.

`sections` is a non-optional property, but the `DataSource` initializer allows us to pass in `nil` for `sections`. We're just defaulting to `[]` in the case where `nil` is passed in.

If we did a fall through on the if `oldSections` is empty, the first time the table view is loaded, the sections would animate in, which likely wouldn't be the desired behaviour.

If that makes sense, I'll make the changes allowing `sections` to be an optional property.

Reply:
Looking at this again, I think you're right with the fall through. If we're setting the sections for the first time, or if the section count hasn't changed, we just call `reloadSections` and nothing would be animated. Tested this out with the example project and things were looking good.

Reply:
Do you think we still need the is empty check? Which determines reloadData() call?

Reply:
Removing the `isEmpty` check and relying solely on the add/remove/update/reload animations is throwing some memory errors when I run the tests.

Given that the initializer allows for nil sections to be passed in, part of me is thinking it would be more explicit to work with nullable sections, like we also handle the case where `tableView` is nil.